apps/idos-enclave/src/lib/enclave.js

```diff
@@ -305,6 +305,24 @@ export class Enclave {
       .filter(({ content }) => negate(() => matchCriteria(content, privateFieldFilters.omit)));
   }
 
+  async discoverUserEncryptionKey() {
+    // biome-ignore lint/suspicious/noAsyncPromiseExecutor: <explanation>
+    return new Promise(async (resolve, reject) => {
+      try {
+        const { password } = await this.#openDialog("discoverPublicKey");
+        const humanId = crypto.randomUUID();
+        const secretKey = await idOSKeyDerivation({ password, salt: humanId });
+
+        const keyPair = nacl.box.keyPair.fromSecretKey(secretKey);
+
+        const encryptionPublicKey = Base64Codec.encode(keyPair.publicKey);
+        resolve({ encryptionPublicKey, humanId });
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
```

Comment:
No need to over-wrap this.

```suggestion
  async discoverUserEncryptionKey() {
    const { password } = await this.#openDialog("discoverPublicKey");
    const humanId = crypto.randomUUID();
    const secretKey = await idOSKeyDerivation({ password, salt: humanId });
  
    const keyPair = nacl.box.keyPair.fromSecretKey(secretKey);
  
    const encryptionPublicKey = Base64Codec.encode(keyPair.publicKey);
    return { encryptionPublicKey, humanId };
  }
```